rl_teacher/teach.py

```diff
@@ -54,44 +53,78 @@ def __init__(self, env, summary_writer, comparison_collector, agent_logger, labe
-            dtype=tf.float32, shape=(None, None, self.q_state_size), name="obs_placeholder")
-        self.segment_alt_placeholder = tf.placeholder(
-            dtype=tf.float32, shape=(None, None, self.q_state_size), name="obs_placeholder")
-
-        # A vanilla MLP maps a q_state to a reward
-        self.mlp = FullyConnectedMLP(self.q_state_size)
-        self.q_state_reward_pred = self._predict_rewards(self.segment_placeholder)
-        q_state_alt_reward_pred = self._predict_rewards(self.segment_alt_placeholder)
-
-        # We use trajectory segments rather than individual q_states because video clips of segments are easier for
-        # humans to evaluate
-        segment_reward_pred_left = tf.reduce_sum(self.q_state_reward_pred, axis=1)
-        segment_reward_pred_right = tf.reduce_sum(q_state_alt_reward_pred, axis=1)
+        """
+        Our model takes in path segments with states and actions, and generates Q values.
+        These Q values serve as predictions of the true reward.
+        We can compare two segments and sum the Q values to get a prediction of a label
+        of which segment is better. We then learn the weights for our model by comparing
+        these labels with an authority (either a human or synthetic labeler).
+        """
+        # Set up observation placeholders
+        self.segment_obs_placeholder = tf.placeholder(
+            dtype=tf.float32, shape=(None, None) + self.obs_shape, name="obs_placeholder")
+        self.segment_alt_obs_placeholder = tf.placeholder(
+            dtype=tf.float32, shape=(None, None) + self.obs_shape, name="alt_obs_placeholder")
+
+        # Set up action placeholders
+        if self.discrete_action_space:
+            self.segment_act_placeholder = tf.placeholder(
+                dtype=tf.float32, shape=(None, None), name="act_placeholder")
+            self.segment_alt_act_placeholder = tf.placeholder(
+                dtype=tf.float32, shape=(None, None), name="alt_act_placeholder")
+            # Discrete actions need to become one-hot vectors for the model
+            segment_act = tf.one_hot(tf.cast(self.segment_act_placeholder, tf.int32), self.act_shape[0])
+            segment_alt_act = tf.one_hot(tf.cast(self.segment_alt_act_placeholder, tf.int32), self.act_shape[0])
+        else:
+            self.segment_act_placeholder = tf.placeholder(
+                dtype=tf.float32, shape=(None, None) + self.act_shape, name="act_placeholder")
+            self.segment_alt_act_placeholder = tf.placeholder(
+                dtype=tf.float32, shape=(None, None) + self.act_shape, name="alt_act_placeholder")
+            # Assume the actions are how we want them
+            segment_act = self.segment_act_placeholder
+            segement_alt_act = self.segment_alt_act_placeholder
+
+        # A vanilla multi-layer perceptron maps a (state, action) pair to a reward (Q-value)
+        mlp = FullyConnectedMLP(self.obs_shape, self.act_shape)
```

Comment:
I haven't tested on MuJoCo, as my license expired and I wasn't able to renew. :(